File: Backend/API/Expense/DTO/AddExpenseDto.cs
﻿using System.ComponentModel.DataAnnotations;
using Core.Expense;
using Core.Group;

namespace API.Expense.DTO
{
    public sealed record AddExpenseDto
    {
        [Required, MinLength(1), MaxLength(50)]
        public string Name { get; init; }

        [Required]
        [Range(0, (double)decimal.MaxValue, ErrorMessage = "Only positive number allowed")]
        public decimal Amount { get; init; }

        [Required]
        public Currency Currency { get; init; }

        [Required]
        public ExpenseType Type { get; init; }

        [Required]
        public Guid GroupId { get; init; }

        [Required, MinLength(1)]
        public IReadOnlyList<Deptor> Deptors { get; init; }
    }
}

File: po-rowno-app/src/app/_services/expense/dto/add-expense.dto.ts
import { Currency } from 'src/app/_common/enums/currency.enum';
import { ExpenseType } from 'src/app/_common/enums/expense-type.enum';
import { ExpenseDeptorDto } from './expense-deptor.dto';

export interface AddExpenseDto {
  /**
   * @minLength 1
   * @maxLength 50
   */
  name: string;
  /**
   * @format double
   * @min 0
   */
  amount: number;
  currency: Currency;
  type: ExpenseType;
  /** @format uuid */
  groupId: string;
  /** @minItems 1 */
  deptors: ExpenseDeptorDto[];
}
